api/src/main/java/org/pmiops/workbench/db/dao/UserServiceImpl.java

```diff
@@ -1122,6 +1124,27 @@ public void maybeSendAccessExpirationEmail(DbUser user) {
     rtExpiration.ifPresent(expiration -> maybeSendRegisteredTierExpirationEmail(user, expiration));
   }
 
+  /**
+   * Return a mapping of users to their Annual Access Renewal expiration date for Registered Tier,
+   * for users who have them
+   */
+  @Override
+  public List<UserAccessExpirations> getRegisteredTierExpirations() {
+    // restrict to current RT users
+    return accessTierService.getAllRegisteredTierUsers().stream()
+        .map(u -> ImmutablePair.of(u, getRegisteredTierExpirationForEmails(u)))
```

Comment:
if you don't like `ImmutablePair`, there are many alternatives

Reply:
Pair is not preferred. Also pasted from that java guide:

`The Pair class bundles two heterogeneous object instances into one, with the componentwise equals behavior one would expect.

There is generally little reason to do this for a field, local variable or parameter; just use both constituent values directly and separately.`
I'd suggest to use AutoValue or Map

Reply:
I'm not sure this applies to intermediate stream values.  I don't have the option of using both values as recommended without associating them in some way, so some kind of tuple structure makes the most sense.  Its lifespan is only as long as the next step in the stream pipeline. 

`Map.Entry` would also do the job here but I think `Pair` more clearly expresses what's happening.

`Map` won't work here because it's a single entry.  I'm not familiar with `AutoValue` but it seems to involve code generation?  That does not seem worthwhile.

Reply:
Auto Value is part of McTerra stack that is widely used in McTerra Apps, it's quite useful to generate Java classes: https://docs.google.com/document/d/1ugU-O1s2mh7wfq63kANdhxgU9ph-J6VsCkYjiJv3NlA/edit
One thing I dislike about AoU codebase is using swagger to general all Java classes, and mixed use that on API layer/DB layer/internal service layer.